components/core/src/clp/CurlDownloadHandler.cpp

```diff
@@ -48,13 +58,47 @@ CurlDownloadHandler::CurlDownloadHandler(
     m_easy_handle.set_option(CURLOPT_TIMEOUT, static_cast<long>(overall_timeout.count()));
 
     // Set up http headers
+    constexpr std::string_view cRangeHeaderName{"range"};
+    constexpr std::string_view cCacheControlHeaderName{"cache-control"};
+    constexpr std::string_view cPragmaHeaderName{"pragma"};
+    std::unordered_set<std::string_view> reserved_headers;
+    reserved_headers.insert(cRangeHeaderName);
+    reserved_headers.insert(cCacheControlHeaderName);
+    reserved_headers.insert(cPragmaHeaderName);
```

Comment:
```suggestion
    std::unordered_set<std::string_view> const reserved_headers{
            cRangeHeaderName,
            cCacheControlHeaderName,
            cPragmaHeaderName
    };
```
The general rule is if we can initialize an object on construction, we should do it